server/controller.js

```
@@ -0,0 +1,276 @@
+const yelp = require('yelp-fusion');
+const client = yelp.client(process.env.YELP_APIKEY);
+const db = require('./models/index.js');
+
+exports.searchRestaurants = async (ctx) => {
+  const restaurant = ctx.request.body;
+  try {
+    await client.search(restaurant).then(response => {
+      const results = response.jsonBody.businesses;
+      ctx.body = results;
+    })
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+exports.getListInfo = async (ctx) => {
+  const listId = ctx.params.listId;
+  try {
+    const listInfo = await db.Lists.findOne({
+      where: {
+        id: listId
+      },
+      raw: true
+    })
+    ctx.body = listInfo;
+    ctx.status = 200;
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+// this is only on yumlist
+exports.loadFavoritesFromList = async (ctx) => {
+  const listId = ctx.params.listId;
+  console.log('entered load favs from list and listid is', listId);
+
+  try {
+    const favoritesOnLoad = await db.Favorites.findAll({
+      include: [{
+        model: db.Lists,
+        where: { id: listId }
+      }]
+    });
+    console.log('favorite restaurants in list', favoritesOnLoad);
+    ctx.body = favoritesOnLoad;
+    ctx.status = 200;
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+
+exports.addToFavorites = async (ctx) => {
+  const selectedRestaurant = ctx.request.body;
+  const currentListId = ctx.params.list;
+  
+  // first, get all restaurants in this list
+  const list = await db.Lists.find({
+    where: { id: currentListId },
+    include: [{
+      model: db.Favorites,
+      attributes: ['id'],
+    }]
+  });
+  
+  // then, check if the restaurant exists in this list
+  const restaurantExistsInList = list.dataValues.Favorites.filter(restaurant => restaurant.id === selectedRestaurant.id);
+  console.log('restaurant exists in the list', restaurantExistsInList)
+  
+  if (restaurantExistsInList.length !== 0) {
+    ctx.body = {
+      "error": ['Already added']
+    }
+    ctx.status = 400;
+  } else {
```

To maintain the indentation level as lowest as possible you can do this:

```js
if (restaurantExistsInList.length > 0) {
  ctx.body = {
    "error": ['Already Added']
  }
  ctx.status = 400;
  return;
}
```

Or even better, if you install that Error Handler Middleware I was talking about yesterday, you can throw an Error.